### PR TITLE
Add some helper methods to UnsolicitedMessageHandler.

### LINKED ISSUE #5605: Add a Reset method to UnsolicitedMessageHandler
@andy31415 suggests that a Reset() method may be better than just setting .Delegate to null manually, because it localizes knowledge about what a handler data contents is.

_Originally posted by @andy31415 in https://github.com/project-chip/connectedhomeip/pull/5581#discussion_r599825584_

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
Clarifies some code.

Fixes https://github.com/project-chip/connectedhomeip/issues/5605

### CODE REVIEW

**Review by kghost — APPROVED** [review]

**Review by Damian-Nordic — APPROVED** [review]

File: src/messaging/ExchangeMgr.h [hunk 1]
```diff
@@ -232,8 +232,21 @@ class DLL_EXPORT ExchangeManager : public SecureSessionMgrDelegate, public Trans
     struct UnsolicitedMessageHandler
     {
         UnsolicitedMessageHandler() : ProtocolId(Protocols::NotSpecified) {}
+
+        constexpr void Reset() { Delegate = nullptr; }
+        constexpr bool IsInUse() const { return Delegate != nullptr; }
+        // Matches() only returns a sensible value if IsInUse() is true.
+        constexpr bool Matches(Protocols::Id aProtocolId, int16_t aMessageType) const
```
Comment (andy31415) [inline src/messaging/ExchangeMgr.h:239]: should we return a predictable value is not in use (guessing false?)

so return `IsInUse() && (ProtocolId == aProtocolId) && (MessageType == aMessageType)`

**Review by andy31415 — APPROVED** [review]

File: src/messaging/ExchangeMgr.h (hunk 1 above)
Comment (bzbarsky-apple) [reply src/messaging/ExchangeMgr.h:239]: Yes, we could do that.  I checked, and the codesize is exactly the same, so compilers are doing a good job of optimizing out the redundant IsInUse() checks.

Comment (bzbarsky-apple) [reply src/messaging/ExchangeMgr.h:239]: Actually, I have no idea whether the codesize is the same: none of the code in this file is used so far, so it all gets stripped out of the final binary at the moment.....  Given that, I'd prefer to leave things as-is and then get measurements when they might mean something.

Comment (bzbarsky-apple) [reply src/messaging/ExchangeMgr.h:239]: Filed https://github.com/project-chip/connectedhomeip/issues/5616 to follow up on this once this is not dead code.

**Comment by bzbarsky-apple** [comment]
@msandstedt @woody-apple @jepenven-silabs Please take a look?

**Review by msandstedt — APPROVED** [review]

**Pull request merged**